### ISSUE TITLE
FEATURE REQUEST: Gift Aid Plugin Is Non-Complaint With UK's HMRC

### ISSUE THREAD

**Issue by tictag** [strangerstudios/pmpro-gift-aid#6]
I’ve been looking into Gift Aid and have been working with the UK’s HMRC (Her Majesty’s Revenue and Customs). The short version is that the Gift Aid Add On is non-compliant with HMRC’s guidelines. This is for two principle reasons:

(1) The default Gift Aid Declaration (GAD) the Add On provides is simply incorrect
(2) The method of storing the GAD is not sufficient to provide evidence

Number (1) can easily be rectified by changing the text in the add on (or customised using `gettext` ) but (2) requires a code re-think.

HMRC’s ridiculously long guidelines on processing GADs is [here](https://www.gov.uk/government/publications/charities-detailed-guidance-notes/chapter-3-gift-aid#chapter-37-recording-and-audit-of-gift-aid-declarations).

…but the crux of it is:

> Note: the transcribing of declarations given by donors into a database will not satisfy the requirements to have an auditable record as such a record does not demonstrate that the donor made the declaration.

This is basically what the Gift Aid Add On is doing. It is effectively storing an 'indicator' that a donor made the declaration (i.e. Gift Aid = Yes) but, as the note above suggests, in theory we could modify that indicator.

So in conclusion the Gift Aid Add On is unfortunately just not fit-for-purpose.

There are many software providers that provide complaint solutions listed on HMRC's website [here](https://www.gov.uk/government/publications/charities-online-commercial-software-suppliers/charities-online-commercial-software-suppliers) but this not only means additional cost but also the effort of maintaining two membership databases. Not ideal.

An effort to create an open source solution using PMPro will undoubtedly help out these cash-starved charities.

**Comment by tictag** [comment]
For PMPro moderators, further info [here](https://www.paidmembershipspro.com/forums/topic/feature-request-gift-aid-plugin-is-non-complaint-with-uks-hmrc).

**Comment by tictag** [comment]
**UPDATE**

1. I have contacted a UK association that represents all small charities in the UK with a view to finding out how they currently deal with digital GADs. This should at least provide some solution ideas.
2. I have also requested evaluation versions of various commercial packages that purport to 'handle' UK Gift Aid.

I am hoping that I can use this investigation and analysis to get 'the best of worlds' solution we can then further develop for PMPro.

**Comment by tictag** [comment]
Had a bit of an inspiration today whilst watching a [YouTube video](https://www.youtube.com/watch?v=PAXgXWUruSQ) about electronically signing PDF documents within WordPress.

The problem is that HMRC wants unequivocal and tangible evidence that the giftaider signed the Gift Aid Declaration - and **only the giftaider**. Storing a simple acknowledgement of Gift Aid within a database is **explicitly** prohibited, which is what the PMPro Gift Aid Addon actually does.

So my inspiration is to automagically generate a PDF Gift Aid Declaration (GAD) on membership creation that:

- Is protected from editing
- Contains a unique reference e.g. server timestamp

...which is then:

- Sent to the new member as an attachment to their membership confirmation email and
- Stored against the member's user profile

Because the GAD is tied to a unique event (_e.g. membership creation timestamp_) and cannot be altered (_e.g. by an admin_) this would represent unequivocal evidence that only the member could have created it and that it hasn't been subsequently modified.

**Comment by tictag** [comment]
**REFERENCES**

- Gift Aid Declarations (GAD): https://www.gov.uk/guidance/gift-aid-declarations-claiming-tax-back-on-donations
- GAD Sample Template: https://www.gov.uk/government/publications/charities-gift-aid-declaration-form-for-a-single-donation
- Gift Aid Audit Requirements: https://www.gov.uk/government/publications/charities-detailed-guidance-notes/chapter-3-gift-aid
